text/0000-local-snapshot-file-format/0000-local-snapshot-file-format.md

```diff
@@ -0,0 +1,147 @@
+
+![](https://i.imgur.com/e6WuufK.png)
+
+While the node producing such a full ledger state snapshot could theoretically pre-compute the actual snapshot milestone state, this is deferred to the consumer of the data to speed up local snapshot creation.
+
+A full ledger state local snapshot is denoted by the type byte `0`:
+
+```
+version<byte>
+type<byte> = 0
+timestamp<uint64>
+network_id<uint64>
+seps_milestone_index<uint64>
+ledger_milestone_index<uint64>
+seps_count<uint64>
+outputs_count<uint64>
+diffs_count<uint64>
+seps<array[seps_count]>:
+	sep<array[32]>
+outputs<array[outputs_count]>:
+	transaction_hash<array[32]>
+	output_index<uint16>
+	address:
+		address_type<byte>
+		wots_address<array[49]>
+		||
+		address_type<byte>
+		ed25519_address<array[32]>
+	value<uint64>
+diffs<array[diffs_count]>:
+	diff_milestone_index<array[32]>
+	created_outputs_count<uint64>
+	created_outputs<array>:
+		transaction_hash<array[32]>
+		output_index<uint16>
+		address:
+			address_type<byte> = 0
+			wots_address<array[49]>
+			||
+			address_type<byte> = 1
+			ed25519_address<array[32]>
+		value<uint64>
+	consumed_outputs_count<uint64>
+	consumed_outputs<array>:
+		transaction_hash<array[32]>
+		output_index<uint16>
+		address:
+			address_type<byte>
+			wots_address<array[49]>
+			||
+			address_type<byte>
+			ed25519_address<array[32]>
+		value<uint64>		
+```
+
+#### Delta Ledger State
+
+A delta ledger state local snapshot only contains the `diffs` of milestones starting from a given `ledger_milestone_index`. A node consuming such data must know the state of the ledger at `ledger_milestone_index`.
+
+![](https://i.imgur.com/bt5BUpe.png)
+
+A delta ledger state local snapshot is denoted by the type byte `1`:
+
+```
+version<byte>
+type<byte> = 1
+timestamp<uint64>
+network_id<uint64>
+seps_milestone_index<uint64>
+ledger_milestone_index<uint64>
+seps_count<uint64>
+diffs_count<uint64>
+seps<array[seps_count]>:
+	sep<array[32]>
+diffs<array[diffs_count]>:
+	diff_milestone_index<array[32]>
+	created_outputs_count<uint64>
+	created_outputs<array>:
+		transaction_hash<array[32]>
+		output_index<uint16>
+		address:
+			address_type<byte> = 0
+			wots_address<array[49]>
+			||
+			address_type<byte> = 1
+			ed25519_address<array[32]>
+		value<uint64>
+	consumed_outputs_count<uint64>
+	consumed_outputs<array>:
+		transaction_hash<array[32]>
+		output_index<uint16>
+		address:
+			address_type<byte> = 0
+			wots_address<array[49]>
+			||
+			address_type<byte> = 1
+			ed25519_address<array[32]>
+		value<uint64>
+```
+
+# Drawbacks
+
+Nodes need to support this new format.
+
+# Rationale and alternatives
+
+* In conjunction with a companion full snapshot, a tool or node can "truncate" the data from a delta snapshot back to a single full snapshot. In that case, the `ledger_milestone_index` and `seps_milestone_index` would be the same. In the example above, given the full and delta snapshots, one could produce a new full snapshot for milestone 1350.
```

Comment:
Here you mean that, after the truncation, the `seps_milestone_index` of the Delta will be == to the  `ledger_milestone_index` of the Full. Correct?